ironfish-rust/src/merkle_note.rs

```diff
@@ -89,8 +89,8 @@ impl MerkleNote {
             &note.commitment_point(),
             public_key,
         );
-        let mut note_encryption_keys = [0; NOTE_ENCRYPTION_KEY_SIZE];
-        aead::encrypt(&encryption_key, &key_bytes, &mut note_encryption_keys);
+        let note_encryption_keys: [u8; NOTE_ENCRYPTION_KEY_SIZE] = 
```

Comment:
I didn't want to add a big refactor in here for returning results rather than unwrapping, I can do that in follow up PR if we want to. Also is there any reason why we would be returning plain objects rather than `Result`s?

Reply:
As far as I know, we usually only use `Result`s if we're propagating errors up, so we might not have been doing that before (it doesn't look like the old `crypto-wasm` returned a `Result`, so it probably wasn't necessary for us to return one either)

Reply:
From what I was reading, the result type is idiomatic (functional, strongly typed) rather than unwrap. Takes advantage of the functional nature of rust, which makes me 😄